clients/src/main/java/org/apache/kafka/clients/ManualMetadataUpdater.java

```diff
@@ -69,19 +71,23 @@ public long maybeUpdate(long now) {
     }
 
     @Override
-    public void handleDisconnection(String destination) {
-        // Do nothing
+    public void handleDisconnect(long now, String nodeId, Optional<AuthenticationException> maybeFatalException) {
+        if (maybeFatalException.isPresent()) {
+            // We don't fail the broker on failures, but there should be sufficient information in the
+            // logs indicating the reason for failure.
+            log.error("Received authentication failure from brokerId={}", nodeId, maybeFatalException.get());
```

Comment:
Maybe a debug log level is appropriate here. We discussed this way back in https://github.com/apache/kafka/pull/3832#discussion_r139941391

I imagine this would look something like
```
ERROR [Controller id=0, targetBrokerId=0] Connection to node 0 failed authentication due to: Authentication failed due to invalid credentials with SASL mechanism PLAIN (org.apache.kafka.clients.NetworkClient)
ERROR Received authentication failure from brokerId=0 (org.apache.kafka.clients.ManualMetadataUpdater)
```

Reply:
I think debug logging would be sufficient for this one as well as the log entry below for normal disconnect.

Reply:
In that case, do these log messages have any value? Perhaps we can get rid of them.

Reply:
@rajinisivaram do you know of any cases where the `maybeFatalException.get()` may not be descriptive enough?